src/Export/Services/SalesChannelService.php

```diff
@@ -39,6 +41,7 @@ public function getSalesChannelContext(
             (new Criteria())->addFilter(new EqualsFilter('configurationKey', 'FinSearch.config.shopkey')),
             $currentContext->getContext()
         );
+        $currencyId = $this->findologicConfigService->getDomainCurrencyId($currentContext);
```

Comment:
Relying on the current context could lead to issues, when the shopkey is requested on a different sales channel.
We should fetch the domain within the if block below according to the `languageId` and `salesChannelId`. If no `domainId` is configured, I guess we just take the first one.